### PR TITLE
Enable recursive CTEs in ZetaSQL

### PR DESCRIPTION
**Pull request by fivetran-dmitryboykov**
This PR adds e2e test to check support recursive CTEs.

### CODE REVIEW

**Comment by pgrivachev** [comment]
> This PR enables language feature `FEATURE_V_1_3_WITH_RECURSIVE`. It is needed to support recursive CTEs.

What is CTE?

File: server/src/ZetaSqlWrapper.ts [hunk 1]
```diff
@@ -128,6 +129,7 @@ export class ZetaSqlWrapper {
     // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition -- builtinFunctionOptions from external lib can be null
     if (!this.catalog.builtinFunctionOptions) {
       const languageOptions = await new LanguageOptions().enableMaximumLanguageFeatures();
+      languageOptions.options.enabledLanguageFeatures?.push(LanguageFeature.FEATURE_V_1_3_WITH_RECURSIVE);
```
[the remark this answers is not in the document]
Comment (fivetran-dmitryboykov) [reply server/src/ZetaSqlWrapper.ts:132]: Seems that `enableMaximumLanguageFeatures` enables only that features which marked as `in_development = false`.
But needed feature is 
```
  // Enable WITH RECURSIVE ((broken link))
  FEATURE_V_1_3_WITH_RECURSIVE = 13011
      [(language_feature_options).in_development = true];
```

```
// Enable all optional features and reservable keywords that are enabled in
  // the idealized ZetaSQL and are released to users.
  void EnableMaximumLanguageFeatures() {
    EnableMaximumLanguageFeatures(/*for_development=*/false);
  } 
```

**Comment by fivetran-dmitryboykov** [comment]
> > This PR enables language feature `FEATURE_V_1_3_WITH_RECURSIVE`. It is needed to support recursive CTEs.
> 
> What is CTE?

Common Table Expressions.

**Review by pgrivachev — APPROVED** [review]
👍

**Pull request merged**